sway/desktop/render.c

```diff
@@ -1863,54 +1802,22 @@ void output_render(struct sway_output *output, struct timespec *when,
 
 	int width, height;
 	wlr_output_transformed_resolution(wlr_output, &width, &height);
+
 	if (debug.damage == DAMAGE_RERENDER) {
 		pixman_region32_union_rect(damage, damage, 0, 0, width, height);
 	}
 
-	bool has_blur = false;
-	bool blur_optimize_should_render = false;
 	bool damage_not_empty = pixman_region32_not_empty(damage);
-	pixman_region32_t extended_damage;
-	pixman_region32_init(&extended_damage);
-	if (!fullscreen_con && !server.session_lock.locked && damage_not_empty) {
-		// Check if there are any windows to blur
-		struct workspace_effect_info effect_info = get_workspace_effect_info(output);
-		has_blur = effect_info.container_wants_blur || config->blur_enabled;
-		if (effect_info.should_render_optimized_blur) {
-			blur_optimize_should_render = true;
-			// Damage the whole output
-			pixman_region32_union_rect(damage, damage, 0, 0, width, height);
-		}
-
-		// Extend the damaged region
-		int expanded_size = effect_info.expanded_size;
-		if (expanded_size > 0) {
-			int32_t damage_width = damage->extents.x2 - damage->extents.x1;
-			int32_t damage_height = damage->extents.y2 - damage->extents.y1;
-			// Limit the damage extent to the size of the monitor to prevent overflow
-			if (damage_width > width || damage_height > height) {
-				pixman_region32_intersect_rect(damage, damage, 0, 0, width, height);
-			}
```

Comment:
We should confirm that this isn't an issue, as it fixed a bug from the original PR (#120)

Reply:
I wasn't able to reproduce the original issue, so I'll hold off on resolving this until we have confirmation no fix is needed here.

Reply:
It's still an issue so would be good to still have it added :)

The way that I was able to reproduce it was to have a HDMI monitor plugged into my laptop and quickly disconnect+connect said monitor

Reply:
Something like this should do the trick:
```diff
diff --git a/sway/desktop/render.c b/sway/desktop/render.c
index 1497fac4..86e5c27a 100644
--- a/sway/desktop/render.c
+++ b/sway/desktop/render.c
@@ -1878,9 +1878,17 @@ void output_render(struct sway_output *output, struct timespec *when,
 	} else {
 		bool should_render_blur = should_workspace_have_blur(workspace);
 		if (should_render_blur) {
-			wlr_region_expand(damage, damage, get_blur_size());
-			pixman_region32_copy(&extended_damage, damage);
-			wlr_region_expand(damage, damage, get_blur_size());
+			// Limit the damage extent to the size of the monitor to prevent overflow
+			int32_t damage_width = damage->extents.x2 - damage->extents.x1;
+			int32_t damage_height = damage->extents.y2 - damage->extents.y1;
+			if (damage_width > width || damage_height > height) {
+				pixman_region32_intersect_rect(damage, damage, 0, 0, width, height);
+				pixman_region32_copy(&extended_damage, damage);
+			} else {
+				wlr_region_expand(damage, damage, get_blur_size());
+				pixman_region32_copy(&extended_damage, damage);
+				wlr_region_expand(damage, damage, get_blur_size());
+			}
 		} else {
 			pixman_region32_copy(&extended_damage, damage);
 		}
```

Might need some simplification though 😅